src/components/interaction-designer/flow-editors/FlowEditor.vue

```diff
@@ -75,8 +84,28 @@ const validationVuexNamespace = namespace('validation')
 })
 class FlowEditor extends mixins(Lang) {
   @Prop() readonly flow!: IFlow
-  @Prop({default: 'flow-builder.edit-flow'}) readonly flowHeader!: string
-  @Prop({default: true}) readonly sidebar!: boolean
+  @Prop({default: ''}) readonly flowHeader!: string
+  @Prop({default: true}) readonly isOnSmallContainer!: boolean
+  @Prop({default: false}) readonly didUserSubmit!: boolean
+
+  get hasDefaultInteractionTimeout(): boolean {
+    return !!this.$store.state.trees.ui.appWideInteractionTimeout
+  }
+
+  get firstRowClass(): string {
+    if (this.isOnSmallContainer) {
+      // show full width, no matter we have default InteractionTimeout or not
+      return 'col-12'
+    } if (!this.hasDefaultInteractionTimeout) {
+      return 'col-xl-6 col-lg-6 col-md-12 col-sm-12'
+    } else {
+      return 'col-12'
+    }
+  }
+
+  get otherRowsClass(): string {
+    return this.isOnSmallContainer ? 'col-12' : 'col-xl-6 col-lg-6 col-md-12 col-sm-12'
+  }
```

Comment:
I spent quite some time trying to solve this whole styling thing. I'm glad you succeeded 👏

Reply:
yeah, sometimes it's quite difficult to handle the grid for different container size. I always use bootstrap grid to deal with such AC. https://getbootstrap.com/docs/4.0/layout/grid/#grid-options